providers/certificate.rb

```diff
@@ -52,89 +52,92 @@
   end
 
   if mycert.nil? || mycert.not_after <= renew_at
-    authz = acme_authz new_resource.cn
-
-    case authz.status
-    when 'valid'
-      newcert = acme_cert(new_resource.cn, mykey)
-
-      file "#{new_resource.cn} SSL new crt" do
-        path    new_resource.crt || new_resource.fullchain
-        owner   new_resource.owner
-        group   new_resource.group
-        content new_resource.crt.nil? ? newcert.fullchain_to_pem : newcert.to_pem
-        mode    00644
-        action  :create
-      end
-
-      file "#{new_resource.cn} SSL new chain" do
-        path    new_resource.chain
-        owner   new_resource.owner
-        group   new_resource.group
-        content newcert.chain_to_pem
-        not_if  { new_resource.chain.nil? }
-        mode    00644
-        action  :create
-      end
+    all_validations = [new_resource.cn, new_resource.alt_names].flatten.compact.map do |domain|
+      authz = acme_authz domain
 
-    when 'pending'
-      case new_resource.method
-      when 'http'
-        tokenpath = "#{new_resource.wwwroot}/#{authz.http01.filename}"
+      case authz.status
+      when 'valid'
+        authz
+        next
```

Comment:
I noticed that in case `when 'valid'` doesn't returns validation. I'll update, please wait.